lib/ClangImporter/ImportDecl.cpp

```diff
@@ -3436,12 +3441,19 @@ namespace {
 
       bool hasReferenceableFields = !members.empty();
 
-      if (hasZeroInitializableStorage) {
-        // Add constructors for the struct.
+      const clang::CXXRecordDecl *cxxRecordDecl =
+          dyn_cast<clang::CXXRecordDecl>(decl);
+      if (hasZeroInitializableStorage && !cxxRecordDecl) {
+        // Add default constructor for the struct if compiling in C mode.
+        // If we're compiling for C++, we'll import the C++ default constructor
+        // (if there is one), so we don't need to synthesize one here.
```

Comment:
What do you think about adding an assertion in `synthesizeStructDefaultConstructorBody` to make sure we don't try to synthesize a (likely incorrect) constructor using `Builtin.zeroInitializer`?

Reply:
That seems good.  I wouldn't want us to accidentally miscompile in some weird case.

Reply:
Agreed.